lib/puppet/provider/nexus3_repository/nexus3_repository.rb

```diff
@@ -95,7 +95,7 @@ def canonicalize(context, resources)
       munge_booleans(context, r)
 
       r[:depth] = r[:depth].to_i unless r[:depth].to_s.empty?
-      r[:cleanup_policies].sort!
```

Comment:
Removing the `!` is the same as removing the role line.  
And this would make the result dependent on setting the policies in alphabetical order on your manifest.